Xamarin.Forms.Core/AppThemeColor.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+
+namespace Xamarin.Forms
+{
+	public class AppThemeColor : BindableObject, IDisposable
+	{
+		public AppThemeColor()
+		{
+			Application.Current.RequestedThemeChanged += RequestedThemeChanged;
+		}
+
+		public static readonly BindableProperty LightProperty = BindableProperty.Create(nameof(Light), typeof(Color), typeof(AppThemeColor), default(Color), propertyChanged: (bo, __, ___) => UpdateActualValue(bo));
+
+		public Color Light
+		{
+			get => (Color)GetValue(LightProperty);
+			set => SetValue(LightProperty, value);
+		}
+
+		public static readonly BindableProperty DarkProperty = BindableProperty.Create(nameof(Dark), typeof(Color), typeof(AppThemeColor), default(Color), propertyChanged: (bo, __, ___) => UpdateActualValue(bo));
+
+		public Color Dark
+		{
+			get => (Color)GetValue(DarkProperty);
+			set => SetValue(DarkProperty, value);
+		}
+
+		public static readonly BindableProperty DefaultProperty = BindableProperty.Create(nameof(Default), typeof(Color), typeof(AppThemeColor), default(Color), propertyChanged: (bo, __, ___) => UpdateActualValue(bo));
+
+		public Color Default
+		{
+			get => (Color)GetValue(DefaultProperty);
+			set => SetValue(DefaultProperty, value);
+		}
+
+		private Color _actualValue;
+		public Color ActualValue
```

Comment:
Why `ActualValue` and not just `Value`?

Reply:
At the time I thought it made sense. It can be either the `Light`, `Dark` or `Default` one and this is the _actual_ value that is currently provided by this. But I guess a value is always an actual value :P 

Changed